jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -0,0 +1,259 @@
+    return this;
+  }
+
+  /**
+   * Sets the container entrypoint.
+   *
+   * @param entrypointTokens tokens for the entrypoint command
+   * @return this
+   * @see #setEntrypoint(List) for more details
+   */
+  public JibContainerBuilder setEntrypoint(String... entrypointTokens) {
+    setEntrypoint(Arrays.asList(entrypointTokens));
+    return this;
+  }
+
+  /**
+   * Sets the container entrypoint program arguments. These are additional tokens added to the end
+   * of the entrypoint command.
+   *
+   * <p>For example, if the entrypoint was {@code myprogram --flag subcommand} and program arguments
+   * were {@code hello world}, then the command that run when the container starts is {@code
+   * myprogram --flag subcommand hello world}.
+   *
+   * @param programArguments a list of program argument tokens
+   * @return this
+   */
+  public JibContainerBuilder setProgramArguments(List<String> programArguments) {
+    this.programArguments = ImmutableList.copyOf(programArguments);
+    return this;
+  }
+
+  /**
+   * Sets the container entrypoint program arguments.
+   *
+   * @param programArguments program arguments tokens
+   * @return this
+   * @see #setProgramArguments(List) for more details
+   */
+  public JibContainerBuilder setProgramArguments(String... programArguments) {
+    setProgramArguments(Arrays.asList(programArguments));
+    return this;
+  }
+
+  /**
+   * Sets the container environment. These environment variables are available to the program
+   * launched by the container entrypoint command. This replaces any previously-set environment
+   * variables.
+   *
+   * @param environmentMap a map of environment variable names to values
+   * @return this
+   */
+  public JibContainerBuilder setEnvironment(Map<String, String> environmentMap) {
+    environment = new HashMap<>(environmentMap);
+    return this;
+  }
+
+  /**
+   * Sets a variable in the container environment.
+   *
+   * @param name the environment variable name
+   * @param value the environment variable value
+   * @return this
+   * @see #setEnvironment for more details
+   */
+  public JibContainerBuilder setEnvironmentVariable(String name, String value) {
+    environment.put(name, value);
+    return this;
+  }
+
+  /**
+   * Sets the ports to expose from the container. Ports exposed will allow ingress traffic. This
+   * replaces any previously-set exposed ports.
+   *
+   * @param ports the list of ports to expose
+   * @return this
+   */
+  public JibContainerBuilder setExposedPorts(List<Integer> ports) {
+    this.ports = ports.stream().map(Port::tcp).collect(Collectors.toCollection(ArrayList::new));
+    return this;
+  }
+
+  /**
+   * Sets the ports to expose from the container. This replaces any previously-set exposed ports.
+   *
+   * @param ports the ports to expose
+   * @return this
+   * @see #setExposedPorts(List) for more details
+   */
+  public JibContainerBuilder setExposedPorts(int... ports) {
```

Comment:
Just curious if we won't care about UDP.

Reply:
Yeah, why not use the `Port` class here?

Reply:
I was thinking of not exposing the `Port` class in this interface and keep everything as primitives. (Ideally, we wouldn't expose the `LayerConfiguration` class either.) But, that might just make the API too bloated for setting exposed ports. Will change to just using `Port`.